### PR TITLE
FIO-7498 Fixed the issue with data not being decrypted when trying to…

### PR DESCRIPTION
**Pull request by antonSoftensity**
… export submissions

## Link to Jira Ticket

https://formio.atlassian.net/browse/FIO-7498

## Description

Previously, there were no encryption handlers for the submission exports. Decrypt function was added because there was no way to connect existing encrypt/decrypt behavior to the way data is grabbed from the DB.

## Dependencies

https://github.com/formio/formio-server/pull/1396

## How has this PR been tested?

Tested locally

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] My changes generate no new warnings
- [ ] My changes include tests that prove my fix is effective (or that my feature works as intended)
- [ ] New and existing unit/integration tests pass locally with my changes
- [ ] Any dependent changes have corresponding PRs that are listed above

### CODE REVIEW

**Review by brendanbond — CHANGES_REQUESTED** [review]
1. I'm not fully in context here, but `req.hasSacPackage` and the project secret (see below) are part of the Enterprise Server, whereas we're in OSS Formio. The standard way we deal with that sort of thing is a hook function that is called from OSS which may mutate data based on Enterprise Server features. Is there a hook-based solution we can use rather than an imperative check of the request object? 

2. The project level secret has been [deprecated](https://github.com/formio/formio-server/pull/1274), so it's not a deterministic way to encrypt/decrypt. I'd like us to ensure that the solution to this issue doesn't depend solely on the project secret and falls back to the DB_SECRET; we'll have to think very critically about the potential side effects of the project _not_ containing a secret but the DB_SECRET _not_ being able to decrypt the data in question

**Comment by antonSoftensity** [comment]
> 1. I'm not fully in context here, but `req.hasSacPackage` and the project secret (see below) are part of the Enterprise Server, whereas we're in OSS Formio. The standard way we deal with that sort of thing is a hook function that is called from OSS which may mutate data based on Enterprise Server features. Is there a hook-based solution we can use rather than an imperative check of the request object?
> 2. The project level secret has been [deprecated](https://github.com/formio/formio-server/pull/1274), so it's not a deterministic way to encrypt/decrypt. I'd like us to ensure that the solution to this issue doesn't depend solely on the project secret and falls back to the DB_SECRET; we'll have to think very critically about the potential side effects of the project _not_ containing a secret but the DB_SECRET _not_ being able to decrypt the data in question

I'm not quite sure what is the best solution there (because of the submission data export specifics), but I've changed the way of how data is passed to the OSS, so all the logic is handled in Enterprise now.

File: src/util/decrypt.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
```
Comment (brendanbond) [inline src/util/decrypt.js:1]: This strikes me as risky duplication - if we were to ever need to update enterprise server's encryption/decryption algorithms, we'd have to remember that there are 2 places in which that method is defined (formio-server and OSS formio). Could a better solution here be to encapsulate any decryption logic entirely from the formio-server? One method you might look into is by altering the "exporter" instance to include some sort of decryption subroutine that merely closes over the formio-server decryption fn. Your thoughts?

**Review by brendanbond — CHANGES_REQUESTED** [review]

File: src/util/decrypt.js (hunk 1 above)
Comment (travist) [reply src/util/decrypt.js:1]: We will need to implement a new hook that performs this decryption so we are not duplicating code here.

**Review by travist — CHANGES_REQUESTED** [review]

**Review by brendanbond — APPROVED** [review]

**Pull request merged**